Rubicon-java test suite doesn't run on Apple Silicon

**Describe the bug**

When running the test suite on Apple Silicon, an error is generated because the librubicon.dylib is compiled for arm64 architecture, not x86_64. An x86_64 build is required regardless of the native platform because Java does not currently provide an arm64 build.

**To Reproduce**
Steps to reproduce the behavior:
1. Run the rubicon Java test suite on Apple M1 hardware

**Expected behavior**
 
Test suite should pass. 

**Screenshots**
Running `tox -e py39` yields:
```
...
Exception in thread "main" java.lang.UnsatisfiedLinkError: /Users/rkm/beeware/rubicon/java/build/librubicon.dylib: dlopen(/Users/rkm/beeware/rubicon/java/build/librubicon.dylib, 0x0001): tried: '/Users/rkm/beeware/rubicon/java/build/librubicon.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/local/lib/librubicon.dylib' (no such file), '/usr/lib/librubicon.dylib' (no such file)
        at java.lang.ClassLoader$NativeLibrary.load(Native Method)
        at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1946)
        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1861)
        at java.lang.Runtime.loadLibrary0(Runtime.java:871)
        at java.lang.System.loadLibrary(System.java:1124)
        at org.beeware.rubicon.Python.<clinit>(Python.java:27)
        at org.beeware.rubicon.test.Test.main(Test.java:7)
make: *** [test] Error 1
```

**Environment:**
 - Operating System: macOS
 - Python version: Any
 - Software versions:
   - Rubicon Java 0.2.6

--- a/Makefile
+++ b/Makefile
@@ -47,6 +47,13 @@ ifeq ($(IS_ANDROID),yes)
 	LDFLAGS := $(LDFLAGS) -llog
 endif
 
+# When compiling on macOS, check for M1/Apple Silicon.
+IS_APPLE_SILICON := $(shell uname -ms | grep -q "Darwin arm64" && echo yes || echo no)
+ifeq ($(IS_APPLE_SILICON),yes)
+	CFLAGS := $(CFLAGS) -arch x86_64
+	LDFLAGS := $(LDFLAGS) -arch x86_64
+endif
+
 ifdef JAVA_HOME
 	JAVAC := $(JAVA_HOME)/bin/javac
 else

--- a/changes/72.bugfix.rst
+++ b/changes/72.bugfix.rst
@@ -1,0 +1,1 @@
+Modified test suite to support M1/Apple Silicon.

